crates/adapters/hyperliquid/src/data/mod.rs

```diff
@@ -132,22 +230,51 @@ impl HyperliquidDataClient {
     }
 
     async fn bootstrap_instruments(&mut self) -> Result<Vec<InstrumentAny>> {
-        // TODO: Implement proper instrument conversion from Hyperliquid metadata
-        // For now, return empty list as placeholder
-        let meta = self
-            .http_client
-            .info_meta()
-            .await
-            .context("failed to load meta information")?;
+        let mut instruments = Vec::new();
+
+        // Load perpetual instruments
+        match self.http_client.get_perp_meta().await {
```

Comment:
We can keep this client method higher level as `request_instruments` rather than going for a more direct/inner method, i.e. the calls to the parsing function can go down inside the REST client.